examples/pytorch/gin/multi_gpu_graph_prediction.py

```diff
@@ -0,0 +1,151 @@
+        y_true.append(labels.view(y_hat.shape).detach().cpu())
+        y_pred.append(y_hat.detach().cpu())          
+    y_true = torch.cat(y_true, dim=0).numpy()
+    y_pred = torch.cat(y_pred, dim=0).numpy()
+    input_dict = {'y_true': y_true, 'y_pred': y_pred}
+    return evaluator.eval(input_dict)
+
+def train(rank, world_size, dataset_name, root):
+    dist.init_process_group('nccl', 'tcp://127.0.0.1:12347', world_size=world_size, rank=rank)
+    torch.cuda.set_device(rank)
+
+    dataset = AsGraphPredDataset(DglGraphPropPredDataset(dataset_name, root))
+    evaluator = Evaluator(dataset_name)
+
+    model = GIN(300, dataset.num_tasks).to(rank)
+    model = nn.parallel.DistributedDataParallel(model, device_ids=[rank])
+    optimizer = optim.Adam(model.parameters(), lr=0.001)
+    scheduler = optim.lr_scheduler.StepLR(optimizer, step_size=50, gamma=0.5)
+
+    train_dataloader = DataLoader(
+            dataset[dataset.train_idx], batch_size=32,
+            use_ddp=True, shuffle=True)
+    valid_dataloader = DataLoader(
+            dataset[dataset.val_idx], batch_size=32)
+    test_dataloader = DataLoader(
+            dataset[dataset.test_idx], batch_size=32)
+
+    durations = []
+    for epoch in range(50):
+        model.train()
+        train_dataloader.set_epoch(epoch)
+        t0 = time.time()
+        for batched_graph, labels in train_dataloader:
+            batched_graph, labels = batched_graph.to(rank), labels.to(rank)
+            node_feat, edge_feat = batched_graph.ndata['feat'], batched_graph.edata['feat']
+            logits = model(batched_graph, node_feat, edge_feat)
+            optimizer.zero_grad()
+            is_labeled = labels == labels
+            loss = F.binary_cross_entropy_with_logits(logits.float()[is_labeled], labels.float()[is_labeled])
+            loss.backward()
+            optimizer.step()
+        scheduler.step()
+        tt = time.time()
+
+        if rank == 0:
+            print(tt - t0)
+        durations.append(tt - t0)
+
+        if rank == 0:
+            val_metric = evaluate(valid_dataloader, rank, model, evaluator)[evaluator.eval_metric]
+            test_metric = evaluate(test_dataloader, rank, model, evaluator)[evaluator.eval_metric]
+
+            print(f'Epoch: {epoch:03d}, Loss: {loss:.4f}, '
+                  f'Val: {val_metric:.4f}, Test: {test_metric:.4f}')
+
+        dist.barrier()
+
+    dist.destroy_process_group()
+
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser()
+    parser.add_argument('--dataset', type=str, default="ogbg-molhiv",
+                        choices=['ogbg-molhiv', 'ogbg-molpcba'],
+                        help='name of dataset (default: ogbg-molhiv)')
+    dataset_name = parser.parse_args().dataset
+    root = './data/OGB'
+
+    world_size = torch.cuda.device_count()
+    print('Let\'s use', world_size, 'GPUs!')
+    args = (world_size, dataset_name, root)
+    import torch.multiprocessing as mp
```

Comment:
The dataset should be loaded here as in [the tutorial](https://docs.dgl.ai/tutorials/multi/1_graph_classification.html#sphx-glr-tutorials-multi-1-graph-classification-py).

Reply:
Remove it accidentally, re-add for fix.